rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/ArtifactDownloaderImpl.java

```diff
@@ -26,12 +28,23 @@ public ArtifactDownloaderImpl(ClouddriverService clouddriverService) {
   }
 
   public InputStream downloadArtifact(Artifact artifact) throws IOException {
-    Response response =
-        retrySupport.retry(() -> clouddriverService.fetchArtifact(artifact), 5, 1000, true);
-    if (response.getBody() == null) {
+
+    ResponseBody response =
+        retrySupport.retry(
+            () -> {
+              try {
+                return clouddriverService.fetchArtifact(artifact).execute().body();
+              } catch (IOException e) {
```

Comment:
Let's talk about this.  It would be ideal to handle this more centrally instead of littering every call with a catch block like this.